test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -58,7 +58,13 @@ object Assertion extends AssertionVariants {
     val completeString =
       codeString.flatMap(code =>
         assertionString.map { assertion =>
-          code.blue + " did not satisfy " + assertion.cyan
+          val code0      = code.blue
+          val assertion0 = assertion.cyan
+          new StringBuilder(17 + code0.length + assertion0.length)
+            .append(code0)
+            .append(" did not satisfy ")
+            .append(assertion0)
+            .result()
```

Comment:
We were getting some really weird errors in SN with debug mode in this method. This change didn't really fix it, but since it's more optimized I left it as is